lib/src/main/java/org/team1540/base/util/robots/PIDTuningRobot.java

```diff
@@ -0,0 +1,164 @@
+ * allow the values to take effect. To disable a motor, set its motor ID to -1. Motor 1 will be
+ * configured as the master Talon and motors 2, 3, and 4 will be slaved to it in follower mode.
+ */
+public class PIDTuningRobot extends IterativeRobot {
+
+  @Preference(value = "P", persistent = false)
+  public double p;
+  @Preference(value = "I", persistent = false)
+  public double i;
+  @Preference(value = "D", persistent = false)
+  public double d;
+  @Preference(value = "F", persistent = false)
+  public double f;
+  @Preference(value = "I-Zone", persistent = false)
+  public int iZone;
+  @Preference(value = "Max I-Accum", persistent = false)
+  public int maxIAccum;
+  @Preference(value = "Max Acceleration", persistent = false)
+  public int maxAccel;
+  @Preference(value = "Max Velocity", persistent = false)
+  public int maxVel;
+  @Preference(value = "Setpoint", persistent = false)
+  public double setpoint;
+  @Preference(value = "Invert Sensor", persistent = false)
+  public boolean invertSensor;
+  @Preference(value = "Invert Output", persistent = false)
+  public boolean invertOutput;
+  @Preference(value = "Motor 1 ID", persistent = false)
+  public int motor1ID;
+  @Preference(value = "Motor 2 ID", persistent = false)
+  public int motor2ID;
+  @Preference(value = "Motor 3 ID", persistent = false)
+  public int motor3ID;
+  @Preference(value = "Motor 4 ID", persistent = false)
+  public int motor4ID;
+  @Preference(value = "Enable PID", persistent = false)
+  public boolean enablePID;
+
+  private ChickenTalon motor1 = null;
+  private ChickenTalon motor2 = null;
+  private ChickenTalon motor3 = null;
+  private ChickenTalon motor4 = null;
+
+  private Joystick joystick = new Joystick(0);
+
+  private SendableChooser<ControlMode> controlModeChooser = new SendableChooser<>();
+
+  @Override
+  public void robotInit() {
+    System.out.println("Initializing PID Tuner Robot");
+    System.out.println(
+        "To change the motors to be tuned, change the preference values and then run the Reset command to\n"
+            + " * allow the values to take effect. To disable a motor, set its motor ID to -1. Motor 1 will be \n"
+            + " * configured as the master Talon and motors 2, 3, and 4 will be slaved to it in follower mode.");
+    PreferenceManager.getInstance().add(this);
+    Scheduler.getInstance().run(); // allow the PreferenceManager to update
+
+    controlModeChooser.addDefault("Position", ControlMode.Position);
+    controlModeChooser.addDefault("Velocity", ControlMode.Velocity);
+    controlModeChooser.addDefault("MotionMagic", ControlMode.MotionMagic);
+
+    SmartDashboard.putData("Control Mode Chooser", controlModeChooser);
+    Command reset = new SimpleCommand("Reset", () -> {
+      if (motor1ID != -1) {
+        motor1 = new ChickenTalon(motor1ID);
+      } else {
+        System.err.println("Motor 1 must be set!");
+        return;
+      }
+      if (motor2ID != -1) {
```

Comment:
I want my for loop

Reply:
Can't tho cuz you can't pass a field to assign in and I really really do not want to use arrays. I guess it would be "cleaner" but eh.